### PR TITLE
Feat/war in/update rpc

### PR DESCRIPTION
**Pull request by war-in**
## Usage related changes

- update_block_id function (adds block_number key to block_id if "latest" provided)
- call updated (and its tests)

## Checklist:

- [ ] No linter errors
- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: starknet_devnet/blueprints/rpc.py [hunk 1]
```diff
@@ -173,6 +173,18 @@ def rpc_deploy_transaction(transaction: DeploySpecificInfo) -> RpcDeployTransact
...
+    Updates block_id if it is an instance of str
+    """
+    if isinstance(block_id, str):
+        if block_id == "pending":
+            raise RpcError(code=-1, message="Calls with block_hash == 'pending' are not supported currently.")
+        return {"block_number": state.starknet_wrapper.blocks.get_number_of_blocks() - 1}
+
+    return block_id
+
+
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc.py:187]: I'd add an information somewhere that the default case returns the `latest` block's id.
Also maybe something like `block_id_to_block_number` would describe the function better?
Or maybe a more descriptive docstring would do.

File: starknet_devnet/blueprints/rpc.py [hunk 2]
```diff
@@ -173,6 +173,18 @@ def rpc_deploy_transaction(transaction: DeploySpecificInfo) -> RpcDeployTransact
...
 
 
+def update_block_id(block_id: BlockId) -> BlockId:
+    """
+    Updates block_id if it is an instance of str
+    """
+    if isinstance(block_id, str):
+        if block_id == "pending":
+            raise RpcError(code=-1, message="Calls with block_hash == 'pending' are not supported currently.")
+        return {"block_number": state.starknet_wrapper.blocks.get_number_of_blocks() - 1}
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc.py:183]: What if get_number_of_blocks returns 0?

File: starknet_devnet/blueprints/rpc.py [hunk 3]
```diff
@@ -228,10 +240,10 @@ async def get_state_update(block_id: BlockId) -> dict:
     """
     Get the information about the result of executing the requested block
     """
+    block_id = update_block_id(block_id
+                               )
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc.py:244]: Interesting formatting xD

**Review by drknzz — APPROVED** [review]
🐐

File: starknet_devnet/blueprints/rpc.py (hunk 2 above)
Comment (war-in) [reply starknet_devnet/blueprints/rpc.py:183]: block_number will be -1 and method which use it will throw Invalid block id

**Pull request merged**